redaxo/src/addons/mediapool/package.yml

```diff
@@ -17,7 +17,7 @@ page:
         structure: { title: translate:pool_cat_list,   perm: media/hasAll }
         sync:      { title: translate:pool_sync_files, perm: media/hasAll }
 
-blocked_extensions: [php, php3, php4, php5, php6, php7, phtml, pl, asp, aspx, cfm, jsp, jsf, bat, sh, cgi]
+blocked_extensions: [php, php3, php4, php5, php6, php7, pht, phtml, pl, asp, aspx, cfm, jsp, jsf, bat, sh, cgi, htaccess, htpasswd]
```

Comment:
fehlt hier auch noch `.phar`? oder vllt `.hh` für hack(hhvm) files?

Reply:
fehlt auch `.exe` ?

Reply:
`.js` wg nodejs?

Reply:
Bei phar, hh und exe gehe ich mit. Bei js bin ich mir gerade nicht sicher. Weiß gerade nicht, ob es valide Gründe gibt, ne JS Datei in den Medienpool zu laden?
Ganz früher haben glaube ich manche sogar die JS-Dateien der Seite dort reingeladen, das macht heute vermutlich (hoffentlich) keiner mehr ;)
Von daher kann man sie vielleicht wirklich blockieren.

Reply:
bei `.js`  bin ich mir auch unsicher..

Reply:
Hab mal auf Slack gefragt. @DavidBruchmann sagt:

> angenommen man stellt die Dateien auf der Webseite mit Code-Highlighter dar und macht quasi eine developer-site, wäre das dann sinnvoll?
> der Code wird ja dann nicht ausgeführt...

Finde, ist durchaus ein valider Use case, oder? Keine Ahnung, ob man die dann wirklich in den MP laden würde.

Reply:
jepp lassen wir .js erstmal so wie es ist.

Reply:
Und eine .exe als direkten Download, ist das nicht auch ein valider User case? hmm.
Sind Programme für Windows nicht teilweise direkt die .exe, wenn man sie runterlädt?

Reply:
@IngoWinter meint hingegen, lieber js-dateien auch blocken.

Man kann die blocked_extensions auch programmatisch ändern, wenn man es doch anders braucht:

```php
rex_addon::get('mediapool')->setProperty('blocked_extensions', ...);
```

Reply:
Eigentlich sollte die entsprechende Konfiguration unabhängig von Erweiterungen sein, also nicht nur im AddOn MediaPool

Reply:
Ersetzung von "." durch "_" ist etwas kritisch, da bei mir beides in manchen Verzeichnissen existiert.

Reply:
"manche verzeichnisse"? was meinst du damit? hast du "." im dateinamen? (Im Ordnernamen ist es ja noch erlaubt)

Reply:
sorry, war missverständlich ausgedrückt.
Ich habe manche Dateien die mit "." starten und die gleichen Namen, nur statt "." einen Underscore "_"

Reply:
.exe ist definitiv valid und sollte bleiben. Ich würde nur Serverscripte blockieren, auch .JS Dateien sind wichtig. . in _ umbenennen würde ich nicht machen, dass kann ja wirklich gewollt sein.

Reply:
Hab generel nixx gegen eine Paranoid-Einstellung für Anfänger.
Aber es sollte konfigurierbar sein, ohne Zusatz-AddOn.

Reply:
Kann ein Redakteur Schaden anrichten, indem er eine .js Datei in den Medienpool hochlädt?

Reply:
> Kann ein Redakteur Schaden anrichten, indem er eine .js Datei in den Medienpool hochlädt?

Das frage ich mich auch noch. Selbst wenn NodeJS installiert ist, wird die JS doch nicht durch bloßen Aufruf über den Browser ausgeführt, oder?

Ich habe eben mit Jan beim Mittagessen drüber gesprochen, er ist dafür alle Extensions, die mit PHP oder Apache zu tun haben, zu ergänzen. Also pht, phar, htacccess, htpasswd (und evtl hh).
Aber nicht .exe und .js.

Ich denke, ich würde es auch so machen.

Dann noch zur Umwandlung Punkt in Unterstrich: Das betrifft nur den direkten Upload in den MP, und da nur das erste Zeichen im Dateinamen.
Natürlich kann man trotzdem Dot-Dateien in den Medienpool-Ordner legen. Beim Sync werden die sowieso schon mindestens seit 5.0 nicht zur Auswahl angezeigt.